### PR TITLE
[gym] Remove promotion for use_legacy_build_api

### PR DESCRIPTION
**Pull request by hjanuschka**
lets get rid of the legacy_build_api

### CODE REVIEW

**Comment by ohwutup** [comment]
Hey @hjanuschka, was there some discussion around this that you could reference?

**Comment by KrauseFx** [comment]
I talked in the fastlane slack about this with Helmut, since there are so many people still asking about using the legacy build API, and it's not something we should recommend, since it's deprecated for over 2 years now. It's causing issues with Swift, WatchKit and apps with extensions.

**Comment by KrauseFx** [comment]
We should try to get as little number of legacy build API users as possible and solve the underlying issue instead.

File: gym/lib/gym/detect_values.rb [hunk 1]
```diff
@@ -21,6 +21,11 @@ def self.set_additional_default_values
 
       config[:use_legacy_build_api] = true if Xcode.pre_7?
 
+      if config[:use_legacy_build_api]
+        UI.deprecated("the use_legacy_build_api option is deprecated")
+        UI.deprecated("it should not be used anymore - e.g.: if you use app-extensions")
```
Comment (KrauseFx) [inline gym/lib/gym/detect_values.rb]: Don't we have this message somewhere else already?

File: gym/lib/gym/options.rb [hunk 2]
```diff
@@ -95,6 +95,7 @@ def self.plain_options
                                      is_string: false,
                                      optional: true),
         FastlaneCore::ConfigItem.new(key: :use_legacy_build_api,
+                                     deprecated: 'Do not use this option anymore.',
```
Comment (KrauseFx) [inline gym/lib/gym/options.rb]: Let's replace this with 
```
Don't use this option any more, as it's deprecated by Apple
```

**Review by KrauseFx — CHANGES_REQUESTED** [review]

**Comment by KrauseFx** [comment]
Maybe we should also make the warnings bigger and more prominent? I think we already show the message somewhere.

File: gym/lib/gym/options.rb (hunk 2 above)
Comment (hjanuschka) [inline gym/lib/gym/options.rb]: done

File: gym/lib/gym/detect_values.rb (hunk 1 above)
Comment (hjanuschka) [inline gym/lib/gym/detect_values.rb]: i don't find it anywhere - do you have a hint?

**Review by hjanuschka — COMMENTED** [review]
@KrauseFx feedback addressed
do you have any idea where we can put to this to address the "more prominent" thing?

File: gym/lib/gym/detect_values.rb (hunk 1 above)
Comment (KrauseFx) [inline gym/lib/gym/detect_values.rb]: Never mind

**Review by KrauseFx — APPROVED** [review]
Let's ship this 🎷

**Comment by KrauseFx** [comment]
Could you resolve the 2 merge conflicts, and I can merge 👍

**Comment by hjanuschka** [comment]
@KrauseFx should be mergable 🐙

**Comment by KrauseFx** [comment]
This is a great change, thanks for preparing this  🚀

**Pull request merged**